lib/reference/properties.mjs

```diff
@@ -644,3 +644,146 @@ export const longhandTimeProperties = new Set([
 ]);
 
 export const shorthandTimeProperties = new Set(['transition', 'animation']);
+
+export const pageContextProperties = new Set([
+	'direction',
+	'background-color',
+	'background-image',
+	'background-repeat',
+	'background-attachment',
+	'background-position',
+	'background',
+	'border-top-width',
+	'border-right-width',
+	'border-bottom-width',
+	'border-left-width',
+	'border-width',
+	'border-top-color',
+	'border-right-color',
+	'border-bottom-color',
+	'border-left-color',
+	'border-color',
+	'border-top-style',
+	'border-right-style',
+	'border-bottom-style',
+	'border-left-style',
+	'border-style',
+	'border-top',
+	'border-right',
+	'border-bottom',
+	'border-left',
+	'border',
+	'counter-reset',
+	'counter-increment',
+	'color',
+	'font-family',
+	'font-size',
+	'font-style',
+	'font-variant',
+	'font-weight',
+	'font',
+	'height',
+	'min-height',
+	'max-height',
+	'line-height',
+	'margin-top',
+	'margin-right',
+	'margin-bottom',
+	'margin-left',
+	'margin',
+	'outline-width',
+	'outline-style',
+	'outline-color',
+	'outline',
+	'padding-top',
+	'padding-right',
+	'padding-bottom',
+	'padding-left',
+	'padding',
+	'quotes',
+	'letter-spacing',
+	'text-align',
+	'text-decoration',
+	'text-indent',
+	'text-transform',
+	'white-space',
+	'word-spacing',
+	'visibility',
+	'width',
+	'min-width',
+	'max-width',
+]);
+
+export const marginContextProperties = new Set([
+	'direction',
+	'unicode-bidi',
+	'background-color',
+	'background-image',
+	'background-repeat',
+	'background-attachment',
+	'background-position',
+	'background',
+	'border-top-width',
+	'border-right-width',
+	'border-bottom-width',
+	'border-left-width',
+	'border-width',
+	'border-top-color',
+	'border-right-color',
+	'border-bottom-color',
+	'border-left-color',
+	'border-color',
+	'border-top-style',
+	'border-right-style',
+	'border-bottom-style',
+	'border-left-style',
+	'border-style',
+	'border-top',
+	'border-right',
+	'border-bottom',
+	'border-left',
+	'border',
+	'counter-reset',
+	'counter-increment',
+	'content',
+	'color',
+	'font-family',
+	'font-size',
+	'font-style',
+	'font-variant',
+	'font-weight',
+	'font',
+	'height',
+	'min-height',
+	'max-height',
+	'line-height',
+	'margin-top',
+	'margin-right',
+	'margin-bottom',
+	'margin-left',
+	'margin',
+	'outline-width',
+	'outline-style',
+	'outline-color',
+	'outline',
+	'overflow',
+	'padding-top',
+	'padding-right',
+	'padding-bottom',
+	'padding-left',
+	'padding',
+	'quotes',
+	'letter-spacing',
+	'text-align',
+	'text-decoration',
+	'text-indent',
+	'text-transform',
+	'white-space',
+	'word-spacing',
+	'vertical-align',
+	'visibility',
+	'width',
+	'min-width',
+	'max-width',
+	'z-index',
+]);
```

Comment:
If you import `uniteSets` you could reduce it to:

```suggestion
export const marginContextProperties = uniteSets(pageContextProperties, new Set([
	'unicode-bidi',
	'content',
	'overflow',
	'z-index',
	'vertical-align',
]));
```

I am not sure if that's better to create such a relation though.
i.e. `pageContextProperties` may contain properties that are not part of `marginContextProperties` one day in the future